natlas-server/natlas-db.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+"""
+    This is a special app instance that allows us to perform database operations
+    without going through the app's migration_needed check. Running this script
+    is functionally equivalent to what `flask db` normally does. The reason we
+    can't continue to use that is that command is that it invokes the app instance from
+    FLASK_APP env variable (natlas-server.py) which performs the migration check and exits
+    during initialization.
+"""
+import argparse
+
+from app import create_app
+from config import Config
+from migrations import migrator
+
+parser_desc = "Perform database operations for Natlas. You should always take a backup of your database before you upgrade or downgrade, just in case."
+
+
+def main():
+    parser = argparse.ArgumentParser(description=parser_desc)
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "--upgrade",
+        action="store_true",
+        help="Perform a database upgrade, if necessary",
+    )
+    group.add_argument(
+        "--migrate",
+        action="store_true",
+        help="Development: Create a new database migration, if necessary",
+    )
+    group.add_argument(
+        "--downgrade",
+        action="store_true",
+        help="Revert the most recent database upgrade. Danger: This will destroy data as necessary to revert to the previous version.",
+    )
+    parser.add_argument(
+        "--message", default="", help="Description of migration (--migrate only)"
+    )
```

Comment:
What does a message in a migration do? Where does it go? Why would I want it? Maybe I'm misunderstanding this use case, but isn't the point in that we go from version x to version latest, or can go to latest to version x? Why would I need this script?

Is this script mixing a development concern into a production script? That seems weird to me. If I were a user, I'd be confused even with this help message provided above.

Reply:
Yes, the message is a human readable annotation for the migration script that gets generated. I guess we can have a third script that does exactly the same thing for creating new migrations but gets ignored by the dockerignore, that way it's not exposed to end users.